cloudinit/util.py

```diff
@@ -3242,14 +3251,20 @@ def deprecate(
         f"{deprecated_version} and scheduled to be removed in "
         f"{version_removed}. {message}"
     ).rstrip()
-    if return_log:
-        return deprecate_msg
-    if dedup not in deprecate._log:  # type: ignore
+    if (
```

Comment:
This logic isn't beautiful. With 3.10, I might reach for a `match` statement for more expressive logic, but that's a few years out.

Ideas welcome.

Reply:
Looks fine to me.